src/playermat/Playmat.ttslua

```diff
@@ -93,293 +70,271 @@ function onload(save_state)
+  -- right-click binds to new player color
+  if alt_click then
+    PLAYER_COLOR = color
+    printToColor("Upkeep button bound to " .. color, color)
+    return
+  end
+  
+  local forcedLearning = false
+
+  -- unexhaust cards in play zone, flip action tokens and find forcedLearning
+  for _, v in ipairs(searchArea(PLAY_ZONE_POSITION, PLAY_ZONE_SCALE)) do
+    local obj = v.hit_object
+    if obj.tag == "Card" and not obj.is_face_down and not doNotReady(obj) then
+      local notes = JSON.decode(obj.getGMNotes()) or {}
+      if notes.id == "08031" then
+        forcedLearning = true
+      elseif notes.type ~= "Investigator" then
+        obj.setRotation(PLAY_ZONE_ROTATION)
+
+        -- check for cards with 'replenish' in their metadata
+        if notes.uses ~= nil then
+          local count = notes.uses[1].count
+          local replenish = notes.uses[1].replenish
+          if count and replenish then replenishTokens(obj, count, replenish) end
         end
+      end
+    elseif obj.getDescription() == "Action Token" and obj.is_face_down then
+      obj.flip()
     end
+  end
 
-    local deckSize
-    if drawDeck == nil then
-        deckSize = 0
-    elseif drawDeck.tag == "Deck" then
-        deckSize = #drawDeck.getObjects()
-    else
-        deckSize = 1
-    end
-
-    if deckSize >= numCards then
-        drawCards(numCards)
-        return
+  -- flip investigator mini-card if found
+  -- flip summoned servitor mini-cards (To-Do: don't flip all of them)
+  if activeInvestigatorId ~= nil then
+    local miniId = string.match(activeInvestigatorId, "%d%d%d%d%d") .. "-m"
+    for _, obj in ipairs(getObjects()) do
+      if obj.tag == "Card" then
+        local notes = JSON.decode(obj.getGMNotes())
+        if obj.is_face_down and notes ~= nil and notes.type == "Minicard" and (notes.id == miniId or notes.id == "09080-m") then
+          obj.flip()
+        end
+      end
     end
+  end
 
-    drawCards(deckSize)
-    if discardPile ~= nil then
-        shuffleDiscardIntoDeck()
-        Wait.time(|| drawCards(numCards - deckSize), 1)
-    end
-    printToColor("Take 1 horror (drawing card from empty deck)", messageColor)
-end
+  -- gain a resource
+  RESOURCE_COUNTER.call("add_subtract")
 
-function drawCards(numCards)
-    if drawDeck == nil then return end
-    drawDeck.deal(numCards, PLAYER_COLOR)
-end
-
-function shuffleDiscardIntoDeck()
-    if not discardPile.is_face_down then discardPile.flip() end
-    discardPile.shuffle()
-    discardPile.setPositionSmooth(DRAW_DECK_POSITION, false, false)
-    drawDeck = discardPile
-    discardPile = nil
-end
+  -- gain an additional resource for Jenny Barnes
+  if string.match(activeInvestigatorId, "%d%d%d%d%d") == "02003" then
+    RESOURCE_COUNTER.call("add_subtract")
+    printToColor("Gaining 2 resources (Jenny)", messageColor)
+  end
 
-function patriceDraw()
+  -- draw a card (with handling for Patrice and Forced Learning)
+  if activeInvestigatorId == "06005" then
```

Comment:
Unlikely to happen, but Patrice can actually take Forced Learning.  See the ArkhamDB page for the FAQ - probably better to do nothing if both are present